flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1034,27 +1037,76 @@ struct EmboxCommonConversion : public FIROpConversion<OP> {
     TODO("");
   }
 
+  using StoreOrInserter = std::function<mlir::Value(
+      mlir::Value, llvm::ArrayRef<unsigned>, mlir::Value,
+      const std::function<mlir::Value(mlir::LLVM::LLVMType, mlir::Value)> &)>;
+
+  /// Basic pattern to write a field in the descriptor
+  template <typename BOX>
+  std::tuple<mlir::Value, StoreOrInserter>
+  getStoreOrInsertField(BOX box, OperandTy operands,
+                        mlir::ConversionPatternRewriter &rewriter,
+                        unsigned rank) const {
+    auto loc = box.getLoc();
+    auto firBoxTy = box.getType().template dyn_cast<fir::BoxType>();
+    assert(firBoxTy && "embox must have box type");
+    auto boxPtrTy =
+        this->unwrap(this->lowerTy().convertBoxType(firBoxTy, rank));
+
+    auto *thisBlock = rewriter.getInsertionBlock();
+    if (thisBlock &&
+        mlir::isa<mlir::LLVM::GlobalOp>(thisBlock->getParentOp())) {
+      auto boxTy = boxPtrTy.getPointerElementTy();
+      auto un = rewriter.create<mlir::LLVM::UndefOp>(loc, boxTy);
+      auto *ctx = box.getContext();
+      auto insertField =
+          [&, loc, boxTy,
+           ctx](mlir::Value dest, llvm::ArrayRef<unsigned> fldIndexes,
```

Comment:
Do these variables need to be explicitly listed for the lambda capture here?

Reply:
Changed to `[=]`. I was afraid `[=]` would implicitly copy the *this object which I really did not want here since I have no idea what would happen given this inherits mlir framework classes. `[=,this]` that I wanted to use is C++ 20, but after reading more of cppref, I can see that I am wrong and `[=]` captures a copy of the this pointer, not the `*this` object.

https://en.cppreference.com/w/cpp/language/lambda
>When a lambda captures a member using implicit by-copy capture, it does not make a copy of that member variable: the use of a member variable m is treated as an expression (*this).m, and *this is always implicitly captured by reference: